Added dark mode + improved design

kamil-matula opened the pull request:
**DONE**:
1. Added darker colors for details' status bars.
2. Exported all remaining colors used in application to theme.js (tabs, cards, captions and backgrounds).
3. Added colors for Dark Theme basing on Figma.
4. Modified project files to use theme.js fully.
5. Added ThemeDataContext to store the state of current theme.
6. Added onPress attribute to third AppBar's button.
7. Added setting status bar style to "light" so everything is visible on blue background and on black background.
8. Fixed design in General Tab.
9. Added functions that load and save "dark"/"light" string in Async Storage to use the same display mode in next session.
10. Made sure that sorting menu and filtering are also well-displayed.

**TO DO**:
1. Fix status bar bug (after going back from details, it blinks once).
2. Make sure that every pokemon is visible and change the colors if it is too bright or too dark (in Cards and in DetailsAppBar).
3. Consider making the chips brigther (maybe even white background + dark font) in the Dark Mode.

**SCREENSHOTS**:

![image](https://user-images.githubusercontent.com/43967269/119232995-44f5a500-bb27-11eb-872e-570b582c3542.png)
![image](https://user-images.githubusercontent.com/43967269/119233018-5a6acf00-bb27-11eb-9023-3c0bb18ba3b8.png)
![image](https://user-images.githubusercontent.com/43967269/119233031-69518180-bb27-11eb-9f8f-cb044103266d.png)
![image](https://user-images.githubusercontent.com/43967269/119233047-78d0ca80-bb27-11eb-92e0-3653b9becc91.png)
![image](https://user-images.githubusercontent.com/43967269/119233095-9bfb7a00-bb27-11eb-87d1-aacdcab7c711.png)

kamil-matula committed:
Added Dark Mode + fixed design

DONE:
1. Added darker colors for details' status bars.
2. Exported all remaining colors used in application to theme.js (tabs, cards, captions and backgrounds).
3. Added colors for Dark Theme basing on Figma.
4. Modified project files to use theme.js fully.
5. Added ThemeDataContext to store the state of current theme.
6. Added onPress attribute to third AppBar's button.
7. Added setting status bar style to "light" so everything is visible on blue background and on black background.
8. Fixed design in General Tab.

TO DO:
1. Add storing Theme state on device so it could be still dark after restarting application.
2. Fix status bar bug (after going back from details, it blinks once).
3. Make sure that every pokemon is visible and change the colors if it is too bright or too dark.

kamil-matula committed:
Added storing Theme Settings

Added functions that load and save "dark"/"light" string to use the same display mode in next session.

kamil-matula committed:
Fixed Dark Theme in filtering and sorting

Added theme's captions and backgrounds to filtering and sorting.

Blazevarjo commented on src/components/DetailsAppBar.js:
```diff
@@ -1,22 +1,31 @@
 
+import { ThemeDataContext } from '../contexts/ThemeDataContext';
+
 export default function DetailsAppBar({ color, sprite }) {
+  const { currentTheme } = useContext(ThemeDataContext);
   const colors = useTheme().colors;
   const navigation = useNavigation();
 
-  // For now it is the same color as whole component, but later it will be a little bit darker:
-  setStatusBarBackgroundColor(colors.pokemon.background[color]);
+  setStatusBarBackgroundColor(
+    currentTheme.dark ? colors.primaryDark : colors.pokemon.backgroundDark[color]
+  );
```
`const isDark` = useTheme().dark

Blazevarjo commented on src/contexts/StorageActions.js:
```diff
@@ -26,3 +26,26 @@ export const saveFavouritesIds = async (pokemons) => {
+export const loadThemeSettings = async () => {
+  try {
+    const theme = await AsyncStorage.getItem('themeSettings');
+    if (theme === null) {
+      return null;
+    }
+    return theme;
```
It's equal to
`return theme`

Blazevarjo commented on src/contexts/ThemeDataContext.js:
```diff
@@ -0,0 +1,32 @@
+
+  // Button action: switching theme and saving in Async Storage:
+  const switchTheme = () => {
+    setCurrentTheme(currentTheme == CustomTheme ? CustomDarkTheme : CustomTheme);
```
```suggestion
    setCurrentTheme((theme) => theme == CustomTheme ? CustomDarkTheme : CustomTheme);
```

Blazevarjo reviewed (approved):
GJ 🥇

kamil-matula committed:
Applied some changes after Code Review

1. Changed "currentTheme.dark" to "useTheme().dark" - using Context is not necessary.
2. Removed "currentTheme" from Provider values - it is used only in Provider itself, not in any other file.
3. Removed redundant "if (theme == null) return null" - we need only "return theme".
